### ISSUE TITLE
Adding localization

### ISSUE THREAD

**Issue by sebastianbinder** [MhdHejazi/CoronaTracker#26]
Might be a good idea to prepare this app for localization. No idea how this works on code level, but glad to help out with a German translation as soon as it becomes possible on code level.

**Comment by FluffyDev2019** [comment]
I would gladly help out with a Swedish (Svensk) trasnaltion.

**Comment by MhdHejazi** [comment]
Great. I'll prepare the project for localization.

**Comment by MhdHejazi** [comment]
I prepared the project and added two localization files for:
1. German:
https://github.com/MhdHejazi/CoronaTracker/blob/master/Corona/Localization/de.lproj/Localizable.strings
2. Swedish:
https://github.com/MhdHejazi/CoronaTracker/blob/master/Corona/Localization/sv.lproj/Localizable.strings

The strings need to be translated.

**Comment by sebastianbinder** [comment]
Created a PR #27

**Comment by FluffyDev2019** [comment]
Created a PR #28

**Comment by MhdHejazi** [comment]
Thank you guys for the great work!

BTW, I didn't add country names to the localization files because they're part of the data and may change at any time (happened twice before). Do you think we should also translate them?

**Comment by FluffyDev2019** [comment]
Of course, just point me werr to do that.
Sweden = Sverige
Swedish = Svenska

**Comment by Ale111** [comment]
Hi, I would glad to help out with Italian translation. Can you add the italian localization?

**Comment by MhdHejazi** [comment]
Sure. Here you go:
https://github.com/MhdHejazi/CoronaTracker/blob/master/Corona/Localization/it.lproj/Localizable.strings

**Comment by sebastianbinder** [comment]
Regarding country names: Isn’t there a way to use the Apple Map Kit for that? I think there should be translations somewhere already so you could try to solve this programmatically first. Kinda source says „Germany“ -> query Maps -> getting a country object -> output name in device language. Don’t know if this is possible but might be worth a try.

**Comment by MhdHejazi** [comment]
There is a way to translate country names in iOS. The problem is the non-standard names used in the data (e.g. "Korea, South" vs. "South Korea"). See the country names [here](https://github.com/CSSEGISandData/COVID-19/blob/master/csse_covid_19_data/csse_covid_19_time_series/time_series_19-covid-Deaths.csv).

Anyway, I think I'll try to get the translated names from iOS, and when that fails, use the original name. If the result isn't good enough we can work on normalizing the country names.

**Comment by MhdHejazi** [comment]
I did that. Check the app and let me know how it's looking.

I also translated the month names and rotated the country names in the charts to make sure they don't overlap with each other. If you find any other problem let me know please.

**Comment by ppamorim** [comment]
@MhdHejazi Why the italian translation has the values in english?

https://github.com/MhdHejazi/CoronaTracker/blob/master/Corona/Localization/it.lproj/Localizable.strings

**Comment by sebastianbinder** [comment]
If you read above the Italian file still needs translation 🤷🏼‍♂️

**Comment by ppamorim** [comment]
@sebastianbinder So what's the point to commit the file and setup?

**Comment by sebastianbinder** [comment]
I‘m not speaking Italian. Need to wait for @Ale111 to translate the file. If it’s important for you feel free to translate it yourself.

**Comment by ppamorim** [comment]
So @Ale111 should be the person to open the PR like a did in my translation to Portuguese. https://github.com/MhdHejazi/CoronaTracker/pull/29

**Comment by MhdHejazi** [comment]
Yes, check the replies from above. I added the seed files to simplify the process of translation. With the file added, you don't need Xcode anymore, and you can even do everything from the browser.

**Comment by MhdHejazi** [comment]
I collected all the country names from the data source in [one file](https://github.com/MhdHejazi/CoronaTracker/blob/master/Corona/Resources/ISO-Country.yml) and added the ISO code for each country name.

So all country names are now auto-translated.

**Comment by ppamorim** [comment]
@MhdHejazi Instead of adding wrongly set up translation and fix after, we should add the translations without linking it into the project and a developer with access to Xcode should setup what is needed to embed it in the project.

**Comment by MhdHejazi** [comment]
I understand where you're coming from and agree with you in principle, but I prefer to keep things as simple as possible for those willing to contribute.

**Comment by mhdhejazi** [comment]
We have two new strings to be translated:

https://github.com/mhdhejazi/CoronaTracker/blob/76844e33e3df5ca4431bc81d50ae2db4091ec295/Corona/Localization/Base.lproj/Localizable.strings#L16
https://github.com/mhdhejazi/CoronaTracker/blob/3d8863f709908e292371b32f46996dc39aeab549/Corona/Localization/Base.lproj/Localizable.strings#L21

P.S. The translations can be updated from the web browser if you wish. Just find the proper file [here](https://github.com/mhdhejazi/CoronaTracker/tree/master/Corona/Localization) and press the edit button.

**Comment by dev-sora** [comment]
If you still don't have any 'Japanese' translator, I'll be the one.

**Comment by mhdhejazi** [comment]
@dev-sora Sure! And thank you in advance.

**Comment by dev-sora** [comment]
@mhdhejazi So could you add Japanese directory under Corona/Localization?  Or is it my task?

**Comment by mhdhejazi** [comment]
Sure. Here you go: https://github.com/mhdhejazi/CoronaTracker/blob/master/Corona/Localization/ja.lproj/Localizable.strings

**Comment by dev-sora** [comment]
Hi.  I couldn't push my work on 'master'.  Could you add me as a contributor, or another way?

**Comment by mhdhejazi** [comment]
You need to work on your fork and then [create a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request). Or you can update the file directly in the browser by pressing the edit button, and GitHub will create a fork and a pull request for you when you're done.
<img src="https://user-images.githubusercontent.com/121827/77848840-456b0800-71d0-11ea-8327-e025b2a4734f.png" height="80" />

**Comment by dev-sora** [comment]
Thank you for your kind reply!!  I've already done my work.

**Comment by hdatteln** [comment]
Submitted a PR with Dutch translations: https://github.com/mhdhejazi/CoronaTracker/pull/78

**Comment by mhdhejazi** [comment]
Merged. Thank you!

**Comment by hussamzain** [comment]
Submitted the Arabic language, [#86](https://github.com/mhdhejazi/CoronaTracker/pull/86)

**Comment by mhdhejazi** [comment]
Merged. Thank you!

**Comment by comradekingu** [comment]
I would suggest adding this to the libre software https://weblate.org/hosting/ to make it easier to contribute.(Gratis for libre software as you can see.) Noticed some languages using triple-dot ... instead of ellipis … and various stings missing, which would become more apparent if displayed as "not translated" on the platform.

**Comment by surak** [comment]
Fails to compile, saying that localization is in an invalid format. Go figure. Latest version of everything.

**Comment by mhdhejazi** [comment]
@comradekingu Yeah. I think we have to do that.
@surak I merged a fix for the German translation. Try again now.

**Comment by nikotidar** [comment]
Hi there, I'm from Indonesia. I'm trying to contribute to this app and I'm add Indonesian translation to this app. Maybe you can check this? [#92](https://github.com/mhdhejazi/CoronaTracker/pull/92)

Thanks before 😃

**Comment by mhdhejazi** [comment]
@nikotidar Merged. Thank you for taking the time to contribute.

**Comment by nikotidar** [comment]
> @nikotidar Merged. Thank you for taking the time to contribute.

You're welcome 😃

**Comment by comradekingu** [comment]
https://github.com/mhdhejazi/CoronaTracker/pull/88
Norwegian Bokmål by "Allan Nordhøy" e-mail in my profile. Only use @comradekingu for communication here.

**Comment by mhdhejazi** [comment]
@comradekingu I used the mentions for all contributors as they provide more info. One can see your user card (which contains far more than your name) when they hover over your mention, and they can also open your page and see even more. And I already added the full names and GitHub pages to all translation files.
But if you don't like how your name is mentioned, please feel free to change it and create a PR.
Thank you for your contribution again.

**Comment by mhdhejazi** [comment]
I figured that GitHub doesn't show a user card for mentions in the readme file, so I added the full names for everybody with a link to their profile.

**Comment by mhdhejazi** [comment]
To make it easier to update the translations and contribute, I created a project on poeditor.com and imported all existing translations. From now on, you can easily update the translations from poeditor.com website. No need for manual updates and pull requests anymore.

The project link: https://poeditor.com/join/project/UTZyE0GKiT

**Comment by KovalevArtem** [comment]
![image](https://user-images.githubusercontent.com/36500228/80633845-1ca19100-8a62-11ea-93d3-f26a84301acf.png)
Not sure if these lines require translation

**Comment by mhdhejazi** [comment]
Just reload the page. The words are added.
